src/Compilers/CSharp/Test/Emit2/Semantics/ExperimentalAttributeTests.cs

```diff
@@ -0,0 +1,910 @@
+        Assert.Equal("DiagID1", diag.Id);
+        Assert.Equal(ErrorCode.WRN_Experimental, (ErrorCode)diag.Code);
+        Assert.Equal(DefaultHelpLinkUri, diag.Descriptor.HelpLinkUri);
+    }
+
+    [Fact]
+    public void NullUrlFormat_Metadata()
+    {
+        var libSrc = """
+[System.Diagnostics.CodeAnalysis.Experimental("DiagID", UrlFormat = null)]
+public class C
+{
+    public static void M() { }
+}
+
+namespace System.Diagnostics.CodeAnalysis
+{
+    [AttributeUsage(AttributeTargets.All, Inherited = false)]
+    public sealed class ExperimentalAttribute : Attribute
+    {
+        public ExperimentalAttribute(string diagnosticId) { }
+        public string UrlFormat { get; set; }
+    }
+}
+""";
+
+        var libComp = CreateCompilation(libSrc);
+
+        var src = """
+C.M();
+""";
+
+        var comp = CreateCompilation(src, references: new[] { libComp.EmitToImageReference() });
+        comp.VerifyDiagnostics(
+            // (1,1): warning DiagID: 'C' is for evaluation purposes only and is subject to change or removal in future updates.
+            // C.M();
+            Diagnostic("DiagID", "C").WithArguments("C").WithLocation(1, 1)
+            );
+
+        var diag = comp.GetDiagnostics().Single();
+        Assert.Equal(DefaultHelpLinkUri, diag.Descriptor.HelpLinkUri);
+    }
+
+    [Theory, CombinatorialData]
+    public void Suppressed(bool inSource)
+    {
+        var libSrc = """
+[System.Diagnostics.CodeAnalysis.Experimental("DiagID1")]
+public class C
+{
+    public static void M() { }
+}
+""";
+
+        var src = """
+#pragma warning disable DiagID1
+C.M();
+""";
+
+        var comp = inSource
+            ? CreateCompilation(new[] { src, libSrc, experimentalAttributeSrc })
+            : CreateCompilation(src, references: new[] { CreateCompilation(new[] { libSrc, experimentalAttributeSrc }).EmitToImageReference() });
+
+        comp.VerifyDiagnostics();
+    }
+
+    [Theory, CombinatorialData]
+    public void InObsoleteMethod(bool inSource)
+    {
+        // Diagnostics for [Experimental] are not suppressed in [Obsolete] members
+        var libSrc = """
+public class C
+{
+    [System.Diagnostics.CodeAnalysis.Experimental("DiagID1")]
+    public static void M() { }
+}
+""";
+
+        var src = """
+class D
+{
+    [System.Obsolete("obsolete", true)]
+    void M2()
+    {
+        C.M();
+    }
+}
+""";
+
+        var comp = inSource
+            ? CreateCompilation(new[] { src, libSrc, experimentalAttributeSrc })
+            : CreateCompilation(src, references: new[] { CreateCompilation(new[] { libSrc, experimentalAttributeSrc }).EmitToImageReference() });
+
+        comp.VerifyDiagnostics(
+            // (6,9): warning DiagID1: 'C.M()' is for evaluation purposes only and is subject to change or removal in future updates.
+            //         C.M();
+            Diagnostic("DiagID1", "C.M()").WithArguments("C.M()").WithLocation(6, 9)
+            );
+    }
```

Comment:
>} [](http://example.com/codeflow?start=4&length=1)

Consider using a method or type marked with `[Experimental("ID1")][Obsolete("message", true)]`, where the symbol is in source or metadata.